asyncio_mqtt/client.py

```diff
@@ -104,16 +105,17 @@ def __init__(
 )
 
 P = ParamSpec("P")
+S = TypeVar("S", bound="Client")
 
 
 # TODO: Simplify the logic that surrounds `self._outgoing_calls_sem` with
 # `nullcontext` when we support Python 3.10 (`nullcontext` becomes async-aware in
 # 3.10). See: https://docs.python.org/3/library/contextlib.html#contextlib.nullcontext
 def _outgoing_call(
-    method: Callable[Concatenate[Client, P], Coroutine[Any, Any, T]]
-) -> Callable[Concatenate[Client, P], Coroutine[Any, Any, T]]:
+    method: Callable[Concatenate[S, P], Coroutine[Any, Any, T]]
+) -> Callable[Concatenate[S, P], Coroutine[Any, Any, T]]:
     @functools.wraps(method)
-    async def decorated(self: Client, *args: P.args, **kwargs: P.kwargs) -> T:
+    async def decorated(self: S, /, *args: P.args, **kwargs: P.kwargs) -> T:
```

Comment:
Remove the `/`. PEP 570 (positional-only arguments) is pythton 3.8 and forward. We have to support python 3.7 for now.

Reply:
Done.